streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -474,20 +470,17 @@ public void update(final Set<TopicPartition> topicPartitions, final Map<String,
 
     @Override
     public void closeAndRecycleState() {
-        suspend();
-        prepareCommit();
-        writeCheckpointIfNeed();
-
+        // Stream tasks should have already been suspended and their consumed offsets committed before recycling
```

Comment:
Is the comment necessary? Seem the code is self-explaining?

Reply:
Yeah it does seem unnecessary